reference_guide/custom_language_support/references_and_resolve.md

```diff
@@ -26,7 +26,7 @@ The main method of the
 [`PsiReference`](upsource:///platform/core-api/src/com/intellij/psi/PsiReference.java)
 interface is `resolve()`, which returns the element to which the reference points, or `null` if it was not possible to resolve the reference to a valid element (for example, should it point to an undefined class). The resolved element should implement the [`PsiNamedElement`](upsource:///platform/core-api/src/com/intellij/psi/PsiNamedElement.java) interface.
 
-> **NOTE** While both the referencing element and the referenced element both have a name, only the element which **introduces** the name (e.g. the definition `int x = 42`) needs to implement the [`PsiNamedElement`](upsource:///platform/core-api/src/com/intellij/psi/PsiNamedElement.java) interface. It is not necessary for the referencing element at the point of usage (e.g. the `x` in the expression `x + 1`) to implement `PsiNamedElement`.
+> **NOTE** While it may occure that the referencing element and the referenced element both have a name, only the element which **introduces** the name (e.g. the definition `int x = 42`) should implement the [`PsiNamedElement`](upsource:///platform/core-api/src/com/intellij/psi/PsiNamedElement.java) interface. The referencing element at the point of usage (e.g. the `x` in the expression `x + 1`) should not implement `PsiNamedElement` since it itself does not _have_ a name.
```

Comment:
occure -> occur

Reply:
"should [not]" -> "must [not]" (2 times)?

Reply:
They both are not a "must":
- not all declarations are PsiNamedElement-s, one can resolve to PomTargetPsiElement. 
- some references might actually declare a new entity, like reference expressions in JavaScript.